File: lab1/print_lines.pl
#!/usr/bin/perl
#Author: Ryosuke Morino
#

if (@ARGV != 3)
{
	die "usage: print_lines.pl start_line stop_line file\n";
}

$start = $ARGV[0];
$end = $ARGV[1];
$file = $ARGV[2];

if ( $end < $start )
{
	die "start_line must be <= to stop_line\n";
}

else
{
open($IN, "<$file") or die "couldn't open $file";
$count = 1;
	while($line = <$IN>)
	{
		if ($count >= $start and $count <= $end )
		{
			chomp($line);
			print "[$line]\n";
			$count++;		
		}
		else
		{
			$count++;
		}
	}
	if ($count < $start or  $count < $end)
	{
		$count = $count - 1;
		print "ran out of lines, stopping at line $count\n";
	}
close ($IN);
}

File: lab3/print_lines.py
#!/usr/bin/python

import sys

if len(sys.argv) != 4:
    print "usage: print_lines.py start_line stop_line file"
    sys.exit()

start = int(sys.argv[1])
end = int(sys.argv[2])


if end < start:
    print "start_line must be <= to stop_line"
    sys.exit()


else:
    try:
        file = open(sys.argv[3], "r")
        line = file.readline()
    except IOError:
        print "could'nt open", sys.argv[3]
        sys.exit()

    count = 1
    while line:
        if count >= start and count <= end:
            line = line.rstrip("\n")
            print line
            line = file.readline()
            count+=1
        else:
            count+=1
            line = file.readline()
    if count < start or count < end:
        print "ran out of lines stoping at line", count
    file.close()
